src/main/telemetry/ibus.c

```diff
@@ -277,12 +282,13 @@ static void dispatchMeasurementRequest(ibusAddress_t address) {
 
     case IBUS_SENSOR_TYPE_TEMPERATURE:
         {
+            //TODO: handle arithmetric better
             #ifdef BARO
-                float temperature = (baroTemperature + 50) / 100.f;
+                uint16_t temperature = (baroTemperature + 50) / 10;
```

Comment:
`(baroTemperature + 5) / 10` or just `baroTemperature / 10`
And type is `int16_t` (or even better `int`, gcc will generate better code)

It was probably used to round correctly (2160  centidegrees to 22 degrees). But it will return invalid results for negative values (division rounds toward zero). 0.1C is much smaller that sensor accuracy, it's OK to ignore rounding .. 
This code will round correctly, but is a bit of overkill:
```
temperature = baroTemperature / 5;     // divide in two steps
temperature = (temperature + 1) >> 2; // left shift rounds toward negative infinity
```